complex/src/lib.rs

```diff
@@ -33,8 +33,34 @@ use traits::{Zero, One, Num, Float};
 // probably doesn't map to C's _Complex correctly.
 
 /// A complex number in Cartesian form.
+///
+/// ## Representation and Foreign Function Interface Compatibility
+///
+/// `Complex<T>` is currently marked `#[repr(C)]` which means that it is *memory
+/// layout* compatible with an array `[T; 2]` as well as C's complex floating
+/// point types.
+///
+/// Note that `Complex<F>` where F is a floating point type is **only** memory
+/// layout, not necessarily **calling convention** compatible. This means for
+/// ffi you can only pass `Complex<F>` behind a pointer, not as a value.
```

Comment:
I appreciate keeping this simple, that we won't make any guarantees about calling convention.  I had trouble parsing it though -- the gap from "memory layout" to "compatible" is a bit awkward.  Alternates:

- add a comma: "is only memory layout, not necessarily calling convention, compatible."
- repeated: "is only memory layout compatible, not necessarily calling convention compatible."
- reordered: "is only compatible in memory layout, not necessarily calling convention."

I think I prefer it reordered.

Also, I'd apply symmetry to the bold style, either "**only**"/"**not**" or "**memory layout**"/"**calling convention**".

These are nits, but we're trying to clarify an important point here.